chapters/packages.tex

```diff
@@ -326,3 +326,116 @@ \section{External resources}\label{external-resources}
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+\end{lstlisting}
+All general terms in the header should be replaced by specific information.
+
+In the body, directly following the header, for each descriptive text there is an entry of the form
+\begin{lstlisting}
+#: <file name>:<line number>
+msgctxt "<context identifier>"
+msgid "<descriptive text>"
+msgstr "<translation>"
+\end{lstlisting}
+Only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used.
+
+At first there can be an optional comment describing the location (file name and line number) of the text to translate. Multiple occurences of the same string can be listed here, separated by space.
+
+Then, the \lstinline!<context identifier>! behind the keyword \lstinline!msgctxt! is the full name of the Modelica class (e.g. \lstinline!Modelica.Blocks.Math.Sin!) where the text appears in.
+Short class definitions do not appear here.
+Texts in such classes belong to the enclosing full class definition.
+
+After the `msgid` keyword the text string which shall be translated follows.
+It should contain the original string from the Modelica text representation.
+Since in Modelica control sequences also start with a backslash and another backslash is used to use sequences literally or to hide double quotes, no change is required here.
+But Modelica allows strings to go over more than one line, gettext does not.
+Therefore, line breaks in Modelica must be encoded with \lstinline!"\n"! for gettext.
+
+\begin{nonnormative}
+In order to avoid long lines, strings may be split into smaller parts given on consecutive lines, e.g.
+\begin{lstlisting}
+"A
+B"
+\end{lstlisting}
+is converted to
+\begin{lstlisting}
+"A\n"
+"B"
+\end{lstlisting}
+These strings are concatenated when read.
+Please regard that if a \lstinline!msgid! string is given more than once in the same context, all occurrences are translated with the same (last) translation!
+\end{nonnormative}
+
+The keyword \lstinline!msgstr! is followed by the translation of \lstinline!msgid!.
+With regard to control characters and line breaks, the same rules apply as for msgid.
+In the template file this string is empty by definition.
+If this is empty in a language specific file the contents of \lstinline!msgid! may be used instead.
+
+The texts in following Modelica constructs should be translated:
+\begin{itemize}
+\item description strings of component declarations and classes
+\item strings in following annotations:
+\begin{itemize}
+\item Text.string, Text.textString
+\item missingInnerMessage, obsolete, unassignedMessage
+\item Dialog.[group|tab]
+\item Dialog.[load|save]Selector.[caption|filter]
+\item Documentation.[info|revisions]
+\item Figure.title, Plot.title, Curve.legend
```

Comment:
Formatting (there might be more compact ways to write, but I'll not try to come up with something right now):
```suggestion
\item \lstinline!Text.string!, \lstinline!Text.textString!
\item \lstinline!missingInnerMessage!, \lstinline!obsolete!, \lstinline!unassignedMessage!
\item \lstinline!Dialog.group!, \lstinline!Dialog.tab!
\item \lstinline!Dialog.loadSelector.caption!, \lstinline!Dialog.loadSelector.filter!, \lstinline!Dialog.saveSelector.caption!, \lstinline!Dialog.saveSelector.filter!
\item \lstinline!Documentation.info!, \lstinline!Documentation.revisions!
\item \lstinline!Figure.title!, \lstinline!Plot.title!, \lstinline!Curve.legend!
```